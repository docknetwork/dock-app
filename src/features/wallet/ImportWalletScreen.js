import React from 'react';
import {
  Button,
  Header,
  Footer,
  Content,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {BackButton} from '../../design-system/buttons';
import {useDispatch} from 'react-redux';
import {walletOperations} from './wallet-slice';
import {WalletConstants} from './constants';
import {translate} from '../../locales';

export function ImportWalletScreen({onSubmit}) {
  return (
    <ScreenContainer testID="create-wallet-screen">
      <Header>
        <BackButton />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography
          fontFamily="Montserrat"
          fontSize={24}
          lineHeight={32}
          fontWeight="600"
          color="#fff"
          marginTop={52}>
          {
            translate('import_wallet.title')
          }
        </Typography>
        <Typography
          fontSize={16}
          lineHeight={24}
          fontWeight="400"
          marginTop={12}>
          {
            translate('import_wallet.description')
          }
        </Typography>
      </Content>
      <Footer marginBottom={10} marginLeft={26} marginRight={26}>
        <Button
          onPress={onSubmit}
          full
          testID={WalletConstants.importWallet.testID.submitBtn}>
          {
            translate('import_wallet.submit')
          }
        </Button>
      </Footer>
    </ScreenContainer>
  );
}

export function ImportWalletContainer() {
  const dispatch = useDispatch();

  const handleSubmit = () => {
    dispatch(walletOperations.pickWalletBackup());
  };

  return <ImportWalletScreen onSubmit={handleSubmit} />;
}
