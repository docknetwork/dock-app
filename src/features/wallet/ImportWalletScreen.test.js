import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ImportWalletScreen} from './ImportWalletScreen';

const mockStore = configureMockStore();

describe('ImportWalletScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<ImportWalletScreen />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
