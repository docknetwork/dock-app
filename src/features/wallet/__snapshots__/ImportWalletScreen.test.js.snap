// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportWalletScreen should render correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#18181B",
      "flex": 1,
    }
  }
  testID="create-wallet-screen"
>
  <AppGlobalHeader />
  <Styled(View)>
    <BackButton
      accessibilityLabel="BackButton.importWallet"
      testID="BackButton.importWallet"
    />
  </Styled(View)>
  <Styled(ScrollView)
    marginLeft={26}
    marginRight={26}
  >
    <Styled(Component)
      marginTop={52}
      onPress={[Function]}
      variant="h1"
    >
      Import wallet
    </Styled(Component)>
    <Styled(Component)
      marginTop={12}
    >
      If you backed up your wallet previously, you can upload the file to import your wallet.
    </Styled(Component)>
  </Styled(ScrollView)>
  <Styled(Box)
    marginBottom={10}
    marginLeft={26}
    marginRight={26}
  >
    <Button
      accessibilityLabel="importWallet.submitBtn"
      full={true}
      testID="importWallet.submitBtn"
    >
      Import wallet
    </Button>
  </Styled(Box)>
</RNCSafeAreaView>
`;
